File: java/imagetrack/app/trackobject/imagetrack/app/trackobject/TranslatorAPis/TranslateRetrofit/AppModule.java
package imagetrack.app.trackobject.imagetrack.app.trackobject.TranslatorAPis.TranslateRetrofit;


import dagger.Module;
import dagger.Provides;
import imagetrack.app.trackobject.Constants.Constants;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

@Module(includes = Authmodule.class)
public class AppModule {



    @Provides
    Retrofit  provideRetrofit(){
        return new Retrofit.Builder().baseUrl(Constants.BASE_URL).addCallAdapterFactory(RxJava2CallAdapterFactory.create()).addConverterFactory(GsonConverterFactory.create()).build();
    }





}

File: admin/src/main/java/image/crystalapps/ekommerceAdmin/data/AppModule.kt
package image.crystalapps.ekommerceAdmin.data

import android.app.Application
import dagger.Module
import dagger.Provides
import image.crystalapps.ekommerceAdmin.Utils.NotificationUtils.NotificationUtils.BASEURL
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

@Module
  class AppModule {

  @CustomScope
  @Provides
  fun getDataManager(iCloudNotification : ICloudNotification) : DataManager { return AppDataManager(iCloudNotification) }


  @CustomScope
  @Provides
  fun provideRetrofit(): Retrofit{
    return Retrofit.Builder()
      .baseUrl(BASEURL).addConverterFactory(GsonConverterFactory.create())
      .build()
  }


  @CustomScope
  @Provides
  fun provideMyNotification(retrofit: Retrofit): IMyNotification {
    return retrofit.create(IMyNotification::class.java)
  }
  @CustomScope
  @Provides
  fun provideINotification(iMyNotification: IMyNotification, context: Application): ICloudNotification {
    return CloudNotification(iMyNotification, context)
  }


}